myia/testing/multitest.py

```
@@ -80,9 +81,7 @@ def wrapper(*a, **k):
                 if isinstance(result, AbstractValue):
                     ret_graph = infer_graph(graph, args)
                     ret_type = ret_graph.return_.abstract
-                    assert (
-                        ret_type is result
-                    ), f"Expected {result}, got {ret_type}"
+                    autils.unify(ret_type, result)
```

This could mask errors when ret_type is a Generic.